src/compiler/checker.ts

```diff
@@ -6123,9 +6127,12 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         function symbolToStringWorker(writer: EmitTextWriter) {
             const entity = builder(symbol, meaning!, enclosingDeclaration, nodeFlags)!; // TODO: GH#18217
             // add neverAsciiEscape for GH#39027
-            const printer = enclosingDeclaration?.kind === SyntaxKind.SourceFile ? createPrinter({ removeComments: true, neverAsciiEscape: true }) : createPrinter({ removeComments: true });
+            const [printer, disposePrinter] = enclosingDeclaration?.kind === SyntaxKind.SourceFile

```

Comment:
I used array destructuring here since you can choose the names more easily but IIRC when I last tested, object destructuring was a lot faster.